### PR TITLE
Adding fallback icon functionality #170

### PR DESCRIPTION
**Pull request by plinkpaste**
I've implemented this as described by @devoto13 in #170, however I argue the usefulness of being able to define a global fallback icon, as opposed to specifying a fallback icon for each icon.

It may be that people might want a different fallback icon for different areas of their application.

I'll leave this PR here for review, however.

### CODE REVIEW

**Comment by plinkpaste** [comment]
After some thought on this, you could override the fallback icon config in another more specific module, but I think my point still stands to some extent.

**Comment by devoto13** [comment]
@plinkpaste The idea of this feature was mainly to provide an easy way to deal with the error, when consumer does not want to think or use a more explicit, but verbose solution (like those suggested in https://github.com/FortAwesome/angular-fontawesome/issues/50#issuecomment-519890025).

File: src/lib/icon/icon.component.ts [hunk 1]
```diff
@@ -82,24 +93,20 @@ export class FaIconComponent implements OnChanges {
...
-    private sanitizer: DomSanitizer,
-    private config: FaConfig,
-    private iconLibrary: FaIconLibrary,
-    @Optional() private stackItem: FaStackItemSizeDirective,
-  ) {
-  }
-
   ngOnChanges(changes: SimpleChanges) {
-    if (this.icon == null) {
+    if ((this.icon == null && this.config.fallbackIcon == null)) {
```
Comment (devoto13) [inline src/lib/icon/icon.component.ts]: nit: remove extra pair of `()`

**Comment by plinkpaste** [comment]
> The implementation looks good 👍
> 
> However two more things are required before this PR can be merged:
> 
> * a test case, which shows that this feature works and will ensure that it is not broken in the future
> * revert unrelated formatting changes/code moved around the file (probably done by IDE, similar to the previous PR)

Added two test cases, one to display fallback when icon is missing, and one to display icon when both icon and fallback are specified.

Reverted formatting changes.

Removed double brackets.

**Comment by plinkpaste** [comment]
Well that's weird. Travis build failed on tests, however they pass when I try it:

image

**Comment by devoto13** [comment]
@plinkpaste Looking into it

**Comment by devoto13** [comment]
Not sure what the problem is, tests passed locally for me as well. I’ll check it more later today.

**Comment by devoto13** [comment]
@plinkpaste Travis runs build not on your branch, but on the result of the merge of you branch into the `master`. Please rebase your branch onto the latest master and you should be able to reproduce the failing test locally.

**Comment by plinkpaste** [comment]
Hooray!

**Pull request merged**